h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/strategy/ThresholdTEApplicationStrategy.java

```diff
@@ -0,0 +1,38 @@
+package ai.h2o.targetencoding.strategy;
+
+import water.fvec.Frame;
+import java.util.Arrays;
+
+/**
+ * Strategy that will select only categorical columns with cardinality of greater of equal than specified threshold
+ */
+public class ThresholdTEApplicationStrategy extends TEApplicationStrategy {
+  
+  private Frame _frame;
+  private String[] _excludedColumnNames;
+  private long _threshold;
+
+  /**
+   *  Constructor for selection of categorical columns strategy based on threshold value
+   * @param frame the frame selection is being done from
+   * @param excludedColumnNames the column names we want to exclude from the result 
+   *                            ( i.e. response column for classification tasks , fold column in case it is categorical etc.)
+   * @param threshold categorical columns with higher cardinality than {@code threshold} value will be selected
+   */
+  public ThresholdTEApplicationStrategy(Frame frame,  long threshold, String[] excludedColumnNames) {
+    _frame = frame;
+    _excludedColumnNames = excludedColumnNames; 
+    _threshold = threshold;
+  }
+  
+  public String[] getColumnsToEncode() {
+    return Arrays.stream(_frame.names())
```

Comment:
@seb-h2o @deil87 This is not an ideal design. TE can only be applied to categorical columns and "excludedColumnNames" seem to be repeated in both strategies. In that case, it is better to have one class responsible for filtering the columns and the strategy will just be applied on a column level - that is strategy will have on the input only single column and will only output yes/no for the given column.

Reply:
@michalkurka 
> "excludedColumnNames" seem to be repeated in both strategies

yes, I was thinking about this fact.  I wanted to move it to abstract `TEApplicationStrategy` class to highlight its importance in all possible strategies. But sometimes it feels better when you have field to be defined in the class where it is actually used ( in our case in the constructors of inherited classes). Probably it was not you point here but just wanted to share my thoughts with respect to this parameter. I also had a desire not to duplicate filtering part but having separate class for just filtering is an overhead especially when we don't have other use cases for such a class.

> In that case, it is better to have one class responsible for filtering the columns and the strategy will just be applied on a column level - that is strategy will have on the input only single column and will only output yes/no for the given column

How would you represent for example AllCategoricalsStrategy with suggested design? Abstraction that I had in mind was broader than just "column -> yes/no". We can have ratio strategy for example. We also can select categorical columns to encode based on relative cardinality of them...let's say based on average cardinality. I don't think that we should restrict ourselves with just single column as an input.